src/teams/views/shifts.py

```diff
@@ -329,37 +353,92 @@ def get(self, request: HttpRequest, **kwargs) -> HttpResponseRedirect:
             """,
             )
             messages.error(
-                request,
+                self.request,
                 template.render(Context({"shifts": overlapping_shifts})),
             )
         else:
+            # Remove at most one shift assignment for sale if any
+            # When a shift is for sale and a user presses assign its first assigning the for sale one
+            for shift_assignment in shift.team_members.filter(teamshiftassignment__for_sale=True)[:1]:
+                shift.team_members.remove(shift_assignment)
             shift.team_members.add(team_member)
 
-        kwargs.pop("pk")
+        self.kwargs.pop("pk")
 
-        return HttpResponseRedirect(reverse("teams:shifts", kwargs=kwargs))
+        return HttpResponseRedirect(reverse("teams:shifts", kwargs=self.kwargs))
 
 
-class MemberDropsShift(LoginRequiredMixin, CampViewMixin, View):
+class MemberDropsShift(LoginRequiredMixin, CampViewMixin, UpdateView):
+    model = TeamShift
+    fields = []
+    template_name = "team_shift_confirm_action.html"
+    context_object_name = "shifts"
+    active_menu = "shifts"
     """View for remove a user from a shift."""
-    http_methods = ("get",)
 
-    def get(self, request: HttpRequest, **kwargs) -> HttpResponseRedirect:
+    def get_context_data(self, **kwargs) -> dict[str, object]:
+        """Method for setting the page context data."""
+        context = super().get_context_data(**kwargs)
+        context['action'] = f"Are you sure you want to drop this {self.object}?"
+        context["team"] = Team.objects.get(
```

Comment:
team is available from self.get_object().team

Reply:
the context["team"] is required by the included base file team_base.html